common/src/path.rs

```diff
@@ -1,38 +1,83 @@
 use std::borrow::Cow;
 use std::path::{Path, PathBuf};
 
-fn is_absolute_windows_path(s: &str) -> bool {
-    // UNC
-    if s.len() > 2 && (&s[..2] == "\\\\" || &s[..2] == "//") {
-        return true;
+trait ToChar {
+    fn to_char(self) -> char;
```

Comment:
This trait is introduced to allow abstracting over `u8`, `&u8`, `char` and `&char`. This makes it possible to build generic functions that work in iterators and matchers on both binary slices and strings. Unfortunately, I was not able to find another way to do this with builtin traits like `Borrow` and `Into<char>`.